src/client/reducers/reducer_movies.js

```diff
@@ -0,0 +1,29 @@
+import C from '../actions/constants';
+
+function findIndex(movie, state) {
+    return state.findIndex((item) => {
+        return movie.unit === item.unit;
+    });
+}
+
+export default function (state = [], action) {
+    switch (action.type) {
+        case C.POPULATE_MOVIES:
+            if (Array.isArray(action.payload)) {
+                const filtered = action.payload.filter((item) => {
+                    console.log(findIndex(item, state) < 0);
+                    return findIndex(item, state) < 0;
+                });
+                const merged = [...state, ...filtered];
+                const allIds = merged.map((item) => {
+                    return item.unit;
+                });
+                console.log({ byId: merged, allIds });
+                return { byId: merged, allIds };
+            } return [action.payload];
+        case C.ERROR:
```

Comment:
@vitkuz in some cases it's good to catch this error no middleware/action level and throw another action or show a notification.  There is no specific logic for this case in reducer, right? so we can simply skip it and return state

Reply:
@algbeta Done